### ISSUE TITLE
[tbb] compile error(s) if TBB_USE_DEBUG is defined

### ISSUE THREAD

**Issue by OliverSCE** [msys2/MINGW-packages#13846]
### Description

- compile error(s) if TBB_USE_DEBUG is defined

(TBB_USE_DEBUG  is defined automatically if using cmake via TBBTargets.cmake)

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

Microsoft Windows Version = "10.0.19041.1806"

### MINGW environments affected

- [X] MINGW64
- [ ] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

### Expected behavior

should compile

### Actual behavior

1.)
C:/msys64/mingw64/include/oneapi/tbb/profiling.h:148:36: error: cannot convert 'const char*' to 'const tbb::detail::d0::tchar*' {aka 'const wchar_t*'}
  148 |         r1::itt_set_sync_name(obj, name);

2.) (only in Qt-Projects)
C:/msys64/mingw64/include/oneapi/tbb/profiling.h:229:15: error: expected unqualified-id before ')' token
  229 |     void emit() { }

(name clash with Qt's emit)

### Repro steps

```
echo -e "#include <execution>\nint main(){}" >main.cpp
gcc -DTBB_USE_DEBUG main.cpp -ltbb12
```

**Comment by MehdiChinoune** [comment]
I don't see any reproducer!

**Comment by OliverSCE** [comment]
...updated...

**Comment by MehdiChinoune** [comment]
TBB were not built with `Debug` Build Type, What do you expect?

**Comment by OliverSCE** [comment]
I expect to be able to build an application in debug mode via cmake. As gcc implements the standard parallel C++ features via tbb, there is no other option.

The issue has nothing to do with TBB beeing or not beeing build with DEBUG Type. The compilation fails in the headers, long before any debug-type lib would be linked.

They issue can be easily avoided, if the TBB_USE_DEBUG defines is omitted. However, this define is set by TBBTargets.cmake for any project build in debug mode. 

For me the solution is simple. As I don't need the TBB_USE_DEBUG  functionality at all, I have edited TBBTargets.cmake accordingly. 

I hope the issue becomes clearer now.

**Comment by MehdiChinoune** [comment]
Defining `TBB_USE_DEBUG` with Release libraries is not supported by upstream
see: https://github.com/oneapi-src/oneTBB/blob/v2021.7.0/doc/main/tbb_userguide/Debug_Versus_Release_Libraries.rst

Build your own TBB with `Debug` Build Type.

**Comment by OliverSCE** [comment]
Just to be clear: the issue is related to building in debug mode, not release mode.

To my humble opinion the setting of TBB_USE_DEBUG in TBBTargets.cmake for a debug target is something which should be avoided.

**Comment by MehdiChinoune** [comment]
@OliverSCE I have opened a PR to fix the issue in #13906
Please consider testing the artifacts and see if it fixes the issue.

**Comment by OliverSCE** [comment]
Thanks. 
this fixes issue #1 from above.

Issue #2 (name clash with emit from Qt) is still there.
